src/QueryRenderer.ts

```diff
@@ -423,4 +476,41 @@ class QueryRenderChild extends MarkdownRenderChild {
         }
         return groupingRules.join(',');
     }
+
+    private async getOnClickCallback(
+        task: Task,
+        button: HTMLButtonElement,
+        timeUnit: unitOfTime.DurationConstructor = 'days',
+        amount = 1,
+    ) {
+        const errorMessage = '⚠️ Postponement requires a date: due or scheduled.';
+        if (!task.dueDate && !task.scheduledDate) return new Notice(errorMessage, 10000);
```

Comment:
Why is startDate not handled?
Need to understand and document the behaviour... Which date(s) are changed...